### PR TITLE
MGMT-10311: aci mutation hook

### PR DESCRIPTION
**Pull request by slaviered**
Add a mutating hook to the AgentClusterInstall resource so the default value is set in case the user does not specify a UserManagedNetworking in the Spec.

The default value is `true` in SNO and `false` in case of multi node.

## List all the issues related to this PR

- [x] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Comment by slaviered** [comment]
Manual Tests:
1. Add ACI for SNO. Review the CRD and see that the default value is set correctly
2. Add  ACI for multi-node. Review the CRD and see that the default value is set correctly
3. Add ACI for SNO with wrong value. See that the mutation hook passed the check but the validation hook blocked the operation

File: pkg/webhooks/hiveextension/v1beta1/agentclusterinstall_validation_hook.go [hunk 1]
```diff
@@ -314,12 +314,16 @@ func areImageSetRefsEqual(imageSetRef1 *hivev1.ClusterImageSetReference, imageSe
...
-	return newObject.Spec.ProvisionRequirements.ControlPlaneAgents == 1 &&
-		newObject.Spec.ProvisionRequirements.WorkerAgents == 0 &&
+	return isSNO(newObject) &&
 		newObject.Spec.Networking.UserManagedNetworking != nil &&
 		!*newObject.Spec.Networking.UserManagedNetworking
 }
 
+func isSNO(newObject *hiveext.AgentClusterInstall) bool {
+	return newObject.Spec.ProvisionRequirements.ControlPlaneAgents == 1 &&
+		newObject.Spec.ProvisionRequirements.WorkerAgents == 0
```
Comment (omertuc) [inline pkg/webhooks/hiveextension/v1beta1/agentclusterinstall_validation_hook.go:324]: `newObject.Spec.ProvisionRequirements.ControlPlaneAgents == 1` is enough to determine SNO, as SNO is (starting 4.11) used to refer to single control-plane node clusters.

In the future we may want to have day-1 SNO+workers installation (currently it's only day-2 for adding workers to SNO), so better have this condition be prepared for that

File: pkg/webhooks/hiveextension/v1beta1/agentclusterinstall_mutating_hook.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+				Status: metav1.StatusFailure, Code: http.StatusBadRequest, Reason: metav1.StatusReasonBadRequest,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	// Add the new data to the contextLogger
+	contextLogger.Data["object.Name"] = newObject.Name
+
+	// if UserNetworkManagement is not set by the user apply the defaults:
```
[the remark this answers is not in the document]
Comment (slaviered) [reply pkg/webhooks/hiveextension/v1beta1/agentclusterinstall_mutating_hook.go:114]: Agreed

**Pull request merged**